universal-login-relayer/lib/services/WalletService.js

```diff
@@ -1,11 +1,50 @@
 import WalletContract from 'universal-login-contracts/build/WalletContract';
 import LegacyWallet from 'universal-login-contracts/build/LegacyWallet';
-import {hasEnoughToken, isAddKeyCall, getKeyFromData, isAddKeysCall} from '../utils/utils';
-import {utils, ContractFactory} from 'ethers';
+import {hasEnoughToken, isAddKeyCall, getKeyFromData, isAddKeysCall, sortExecutionsByKey} from '../utils/utils';
+import {utils, ContractFactory, Contract} from 'ethers';
 import defaultDeployOptions from '../config/defaultDeployOptions';
+import {concatenateSignatures, calculateMessageHash} from 'universal-login-contracts';
 
+class PendingExecution {
+
+  constructor(walletAddress, wallet) {
+    this.Wallet = wallet;
+    this.WalletContract = new Contract(walletAddress, WalletContract.interface, this.Wallet);
+    this.CollectedSignatures = [];
+  }
+
+  async getStatus() {
+    return {
+      collectedSignatures: this.CollectedSignatures,
+      totalCollected: this.CollectedSignatures.length,
+      required: await this.WalletContract.requiredSignatures()
+    };
+  }
+
+  async push(from, to, value, data, nonce, gasPrice, gasToken, gasLimit, operationType, signature) {
+    if (collectedSignatures.includes(signature)) {
+      throw 'signature already collected';
+    }
+    const key = await this.WalletContract.getSigner(from, to, value, data, nonce, gasPrice, gasToken, gasLimit, operationType, signature);
+    if (await this.WalletContract.getKeyPurpose(key) === 0) {
+      throw 'invalid signature';
+    }
+    collectedSignatures.push({signature, key});
+  }
+
+  async canExecute() {
+    return collectedSignatures.length >= await this.WalletContract.requiredSignatures();
```

Comment:
extract variable:
`const requireSignatures = this.WalletContract.requiredSignatures()`